.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,48 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '28 5 * * 6'
+
```

Comment:
does this take too long that it can't run on every PR?

Reply:
It's 9 minutes; 2 minutes longer than fedora build and one less than the fuzzer. I'll delete it and we can add it back if we need to.